Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/Saml2AllowNoSubjectAssertionSpecValidator.java

```diff
@@ -26,45 +26,66 @@
  */
 package gov.hhs.fha.nhinc.callback.cxf;
 
-import org.opensaml.saml2.core.Assertion;
-import org.opensaml.xml.validation.ValidationException;
+
+import org.apache.wss4j.common.ext.WSSecurityException.ErrorCode;
+
+import org.apache.wss4j.common.ext.WSSecurityException;
+import org.apache.wss4j.common.saml.SamlAssertionWrapper;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Checks {@link org.opensaml.saml2.core.Assertion} for Spec compliance. This validator relaxes the rules by not
  * checking for the existence of Subject when the Assertion contains AttributeStatements. This is required for
  * interoperability with previous CONNECT gateways.
  */
 public class Saml2AllowNoSubjectAssertionSpecValidator extends Saml2ExchangeAuthFrameworkValidator {
-
+    private static final Logger LOG = LoggerFactory.getLogger(Saml2AllowNoSubjectAssertionSpecValidator.class);
     /**
      * Instantiates a new saml2 allow no subject assertion spec validator.
      */
     public Saml2AllowNoSubjectAssertionSpecValidator() {
         super();
     }
 
+    /* (non-Javadoc)
+     * @see gov.hhs.fha.nhinc.callback.cxf.Saml2ExchangeAuthFrameworkValidator#validateAssertion(org.apache.wss4j.common.saml.SamlAssertionWrapper)
+     */
+    @Override
+    protected void validateAssertion(final SamlAssertionWrapper samlAssertion) throws WSSecurityException {
+        try {
+            validateSubject(samlAssertion.getSaml2());
+        } catch (final WSSecurityException e) {
+            LOG.error("Unable to validate Assertion ", e);
+            throw e;
+        }
+
+    }
+
     /**
      * Checks that the Subject element is present when required, but does not check for the existence of Subject when
      * the Assertion contains AttributeStatements.
      *
      * @param assertion the assertion
      * @throws ValidationException the validation exception
      */
+
     @Override
-    protected void validateSubject(Assertion assertion) throws ValidationException {
+    protected void validateSubject(final Assertion assertion) throws WSSecurityException {
         if ((assertion.getStatements() == null || assertion.getStatements().isEmpty())
                 && (assertion.getAuthnStatements() == null || assertion.getAuthnStatements().isEmpty())
                 && (assertion.getAttributeStatements() == null || assertion.getAttributeStatements().isEmpty())
                 && (assertion.getAuthzDecisionStatements() == null || assertion.getAuthzDecisionStatements().isEmpty())
                 && assertion.getSubject() == null) {
-            throw new ValidationException("Subject is required when Statements are absent");
+            throw new WSSecurityException(ErrorCode.FAILURE,"Subject is required when Statements are absent");
```

Comment:
Formatting suggestion: search this PR for **,"** (i.e., no space between a parameter and a String literal) -- or if you're feeling fancy, you could do a grep for a comma without a space after, to potentially catch additional cases.